standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -2125,13 +2080,12 @@ private ColumnStatisticsObj prepareCSObjWithAdjustedNDV(Object[] row, int i,
         }
       }
     };
-    List<Object[]> list = Batchable.runBatched(batchSize, colNames, b);
 
-    List<ColumnStatistics> result = new ArrayList<ColumnStatistics>(
-        Math.min(list.size(), partNames.size()));
+    List<ColumnStatistics> result = new ArrayList<ColumnStatistics>();
```

Comment:
nit: I think we can use the `partNames().size()` to initialise the size of the list

Reply:
done